NAudio/Wave/WaveOutputs/WasapiOut.cs

```diff
@@ -325,6 +325,25 @@ public void Init(IWaveProvider waveProvider)
                             correctSampleRateFormat = null;
                         }
 
+                        // If still null and we're using exclusive mode, try to get the device format property.
+                        if (correctSampleRateFormat == null && shareMode == AudioClientShareMode.Exclusive)
+                        {
+                            // Based on https://stackoverflow.com/questions/22616924/wasapi-choosing-a-wave-format-for-exclusive-output
+                            byte[] waveFormatBytes = (byte[])mmDevice.Properties[PropertyKeys.PKEY_AudioEngine_DeviceFormat].Value;
+                            if (waveFormatBytes != null)

```

Comment:
I'm not sure if this code is right...

1. Is there an easier way to marshal a byte[] into a WaveFormatExtensible?
2. Is this even correct?  I think PKEY_AudioEngine_DeviceFormat actually retrieves a WAVEFORMATEX, not a WAVEFORMATEXTENSIBLE.  So this may be wrong.